python/unitytrainers/self_play_trainer_controller.py

```diff
@@ -72,22 +79,26 @@ def save_models_and_checkpoints(self, sess, global_step, global_saver):
             checkpoint_path = self.ghost_trainers_path + 'main_brain' + '-' + str(global_step) + '.cptk'
             self.main_trainer_saver.save(sess, checkpoint_path)
 
-    def restore_model_from_checkpoint(self, brain_name, checkpoint_path, sess):
-        self.ghost_trainer_savers[brain_name].restore(sess, checkpoint_path)
-
     def handle_episode_termination(self, curr_info, sess):
         # handle opponent policy change
         super(SelfPlayTrainerController, self).handle_episode_termination(curr_info, sess)
-        raise UnityTrainerControllerException("The handle_episode_termination method was not implemented.")
+        self.elapsed_episodes += 1
+        if self.should_change_ghost_model():
+            for brain_name, saver in self.ghost_trainers.items():
```

Comment:
I would maybe refactor this loop into a method called resample_all_ghosts

Reply:
Sure, will do